include/no_os_pwm.h

```diff
@@ -82,6 +82,10 @@ struct no_os_pwm_init_param {
 	enum no_os_pwm_polarity polarity;
 	/** PWM gpio pin init param*/
 	struct no_os_gpio_init_param *pwm_gpio;
+	/* Timer IRQ ID */
```

Comment:
In no_os_pwm.h we shouldn't care about the specific peripheral that is used to create the pulse width modulated signal. The peripheral could be:
* dedicated PWM peripheral
* Timer with PWM feature (as in your stm32 implementation)
* Timer+Interrupts+GPIO manually handled to create the signal

So the point is, you will need to rename tim_callback to pwm_callback and don't mention anything about a Timer because the implementation on another platform could use a dedicated PWM peripheral for example, and not a Timer...

Reply:
Got your point here. 
I've re-worded the struct members/comments by removing the term 'timer'